type-definitions/tests/merge.js

```diff
@@ -29,6 +29,27 @@ import {
   updateIn,
 } from '../../';
 
+// merge: Objects as Maps
```

Comment:
I did nothing but move this block higher in the file, otherwise it complains that `z` isn't found in `XYPoint` despite `objMap` and `XYPoint` never be used together in the file. It seems like flow somehow retains memory of using `XYPoint` with the `merge()` function and incorrectly applies it to a later use on different input arguments.

@calebmer any idea how to reduce or report this?